connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala

```diff
@@ -216,8 +229,12 @@ private class UCProxy extends TableCatalog with SupportsNamespaces {
     createTable.setName(ident.name())
     createTable.setSchemaName(ident.namespace().head)
     createTable.setCatalogName(this.name)
+
+    val isExternal = properties.containsKey(TableCatalog.PROP_EXTERNAL)
     val storageLocation = properties.get(TableCatalog.PROP_LOCATION)
-    if (storageLocation == null) {
+    val isManagedLocation = Option(properties.get(TableCatalog.PROP_IS_MANAGED_LOCATION))
+      .exists(_.equalsIgnoreCase("true"))
+    if (!isExternal && (storageLocation == null || isManagedLocation)) {
```

Comment:
what happens if `isExternal` is set to true, and yet `(storageLocation == null || isManagedLocation)` is true? That should be illegalstateexception instead of "managed table not supported" as the intent is not clear.

Reply:
From Spark's perspective, this may happen if people do `CREATE EXTERNAL TABLE t ...` without a LOCATION clause. Maybe some catalog implementations treat it as valid by generating location for it, but this will never happen in UC/HMS.

Reply:
What blocks this in UC? If there are no assertions for this, should we add one as part of UCSingleCatalog?

Reply:
makes sense, let me add this check